x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_beats_stats.js

```diff
@@ -97,6 +102,15 @@ export function processResults(results = [], { clusters, clusterHostSets, cluste
         clusters[clusterUuid].module.names = Array.from(moduleSet);
         clusters[clusterUuid].module.count += stateModule.count;
       }
+
+      const stateHost = get(hit, '_source.beats_state.state.host');
+      if (stateHost !== undefined) {
+        const hostSet = clusterArchitectureSets[clusterUuid];
+        const hostArchPlatform = `${stateHost.architecture}/${stateHost.os.platform}`;
```

Comment:
Any reason you didn't just add two fields? (You could use a `Map` to store them by this same key, but then store an object)

Reply:
I can't recall exactly - I think it was a suggestion from @ruflin or @ycombinator (or I made it up). Which would you recommend?

Reply:
Given that they represent different details, I feel like having them separate would be more useful especially as ARM is presumably going to grow in server-side usage.